build/room_ensurer/room_ensurer.py

```diff
@@ -307,32 +328,40 @@ def _ensure_admin_power_levels(self, room_info: RoomInfo) -> None:
 
         merge_dict(current_power_levels, supposed_power_levels)
         try:
-            api.set_power_levels(room_info.room_id, supposed_power_levels)
+            api.set_power_levels(room_id, supposed_power_levels)
         except MatrixError:
-            log.debug("Could not set power levels", room_aliases=room_info.aliases)
+            log.debug("Could not set power levels", room_aliases=room_alias)
 
-    def _create_room(self, server_name: str, room_alias_prefix: str) -> RoomInfo:
-        api = self._apis[server_name]
+    def _create_room(self, room_alias_prefix: str) -> None:
+        api = self._apis[self._own_server_name]
         server_admin_power_levels = self._create_server_user_power_levels()
         response = api.create_room(
             room_alias_prefix,
             is_public=True,
             power_level_content_override=server_admin_power_levels,
         )
 
-        room_alias = f"#{room_alias_prefix}:{server_name}"
-        return RoomInfo(response["room_id"], {room_alias}, server_name)
+        room_alias = f"#{room_alias_prefix}:{self._own_server_name}"
+
+        log.info(
+            "Room created. Waiting for other servers to join.",
+            room_alias=room_alias,
+            room_id=response["room_id"],
+        )
 
     def _connect_all(self) -> Dict[str, GMatrixHttpApi]:
         jobs = {
             gevent.spawn(self._connect, server_name, server_url)
             for server_name, server_url in self._known_servers.items()
         }
-        gevent.joinall(jobs, raise_error=True)
-        log.info("All servers connected")
-        return {server_name: matrix_api for server_name, matrix_api in (job.get() for job in jobs)}
+        gevent.joinall(jobs)
 
-    def _connect(self, server_name: str, server_url: str) -> Tuple[str, GMatrixHttpApi]:
+        return {
+            server_name: matrix_api
+            for server_name, matrix_api in (job.get() for job in jobs if job.get())
+        }
```

Comment:
The change to exclude servers that can't be connected can lead to a `KeyError` in line 108 of `__init__` if it can't connect to our own server.

Reply:
I changed it to throwing an exception if own API is not available because ultimately ensuring the rooms dont make any sense without the own api

Reply:
Yep!